Add Vagrant file for dev environment (#5201)
sunpy/sunpy: docs/dev_guide/contents/newcomers.rst

.. _sunpy repository: https://github.com/sunpy/sunpy
.. _ask here: https://openastronomy.element.io/#/room/#sunpy:openastronomy.org

Setting up a development environment using Vagrant
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The first step is to install `virtualbox`_ and `vagrant`_ .

Reviewer: ```suggestion
If you do not want to install Python to your computer or use a system specific solution, you can use `vagrant`_.

You will need to install `virtualbox`_ and `vagrant`_.
```

Then run the following replacing ``<username>`` with your username.

.. code:: bash

    $ git clone https://github.com/<username>/sunpy.git sunpy-git
    $ cd sunpy-git
    $ vagrant up && vagrant ssh

> This will automatically log you in the VM inside the sunpy-dev conda environment with all the dependencies already installed.

Reviewer: ```suggestion
This will automatically log you in the VM inside a sunpy-dev conda environment with all dependencies pre-installed.
```

.. _sunpy repository: https://github.com/sunpy/sunpy
.. _ask here: https://openastronomy.element.io/#/room/#sunpy:openastronomy.org

> Setting up a development environment using Vagrant (optional)

Reviewer: I think this file should be reverted back to normal. I am worried it might confuse people but if people are aware of vagrant, hopefully they will spot the file in the repo.

Author: Done 👍🏻

> Setting up a development environment using Vagrant (optional)
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
> If you do not want to install Python to your computer or use a system specific solution, you can use `vagrant`_.
> 
> You will need to install `virtualbox`_ and `vagrant`_.
> Then run the following replacing ``<username>`` with your username.
> 
> .. code:: bash
> 
>     $ git clone https://github.com/<username>/sunpy.git sunpy-git
>     $ cd sunpy-git
>     $ vagrant up && vagrant ssh
> 
> This will automatically log you in the VM inside a sunpy-dev conda environment with all dependencies pre-installed.
> 
> .. _virtualbox: https://www.virtualbox.org/manual/ch02.html
> .. _vagrant: https://www.vagrantup.com/docs/installation
> 

Reviewer: ```suggestion
```

Having had some time to think about this, I changed my mind.
Probably should not advertise this just in case it confuses anyone who is new.
